Fix missing submodule in `./x vendor`

ehuss opened the pull request:
The `src/tools/rustc-perf` submodule is needed for vendoring because it is included in the vendor set.

To test this:

1. Get a fresh clone of `rust-lang/rust`
2. `./x vendor`

ehuss committed:
Fix missing submodule in `./x vendor`

The `src/tools/rustc-perf` submodule is needed for vendoring because
it is included in the vendor set.

Kobzol commented on src/bootstrap/src/core/build_steps/vendor.rs:
```diff
@@ -57,7 +57,9 @@ impl Step for Vendor {
 
         // These submodules must be present for `x vendor` to work.
-        for submodule in SUBMODULES_FOR_RUSTBOOK.iter().chain(["src/tools/cargo"].iter()) {
+        for submodule in
```
We already have `default_paths_to_vendor` (used below). Could we reuse that instead of hardcoding a path here?

ehuss replied:
I don't think we can necessarily remove the hard-coded paths for the submodules. I'm assuming your intent is to try to avoid these two getting out of sync? I tried moving them up into `default_paths_to_vendor` in d8e8fc5575ba3f3aa31cb8c7f47df73e06a38e92 to go in that direction.

Kobzol replied:
Yes, sorry, that's what I meant, to make it a bit harder to avoid them going out of sync. This is much nicer, thank you!

jieyouxu commented:
I'm going to r? @Kobzol on this since I don't know the vendoring flow well yet.

ehuss committed:
Keep list of submodules close to list of vendored workspaces

This moves the list of submodules needed to vendor close to the list of
cargo workspaces with the intent to help ensure they keep up-to-date and
in sync.

Kobzol commented:
Thanks for making the change. You can r=me once CI is green.

ehuss commented:
@bors r=kobzol rollup

bors commented:
:pushpin: Commit d8e8fc5575ba3f3aa31cb8c7f47df73e06a38e92 has been approved by `kobzol`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=kobzol d8e8fc5575ba3f3aa31cb8c7f47df73e06a38e92 -->
<!-- homu: {"type":"Approved","sha":"d8e8fc5575ba3f3aa31cb8c7f47df73e06a38e92","approver":"kobzol","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
